packages/core/assemblyManager/assembly.ts

```diff
@@ -312,16 +315,17 @@ async function loadAssemblyReaction(
   const {
     sequenceAdapterConfig,
     assemblyName,
-    assembly,
     refNameAliasesAdapterConfig,
+    pluginManager,
   } = props
 
-  const { pluginManager } = assembly
   const dataAdapterType = pluginManager.getAdapterType(
     sequenceAdapterConfig.type,
   )
-  const adapter = new dataAdapterType.AdapterClass(sequenceAdapterConfig)
-  const adapterRegions = (await adapter.getRegions()) as Region[]
+  const adapter = new dataAdapterType.AdapterClass(
+    sequenceAdapterConfig,
+  ) as RegionsAdapter
+  const adapterRegions = (await adapter.getRegions({ signal })) as Region[]
```

Comment:
This is a nice catch about signal. In the future where we want to access with authorization, how we can pass a more generic set of flags here. statusCallback is another example